pkg/migration/cleanup.go

```diff
@@ -258,6 +260,9 @@ func (h *Handler) deleteMonitoringComponent(cluster *corev1.StorageCluster) erro
 	if err := h.deleteObject(prometheusOpAccountName, cluster.Namespace, &v1.ServiceAccount{}, cluster); err != nil {
 		return err
 	}
+	if err := h.deleteObject(component.TelemetryConfigMapName, cluster.Namespace, &v1.ConfigMap{}, cluster); err != nil {
```

Comment:
there is no need to delete it, the configMap will be reused by the v1 telemetry. if it's v2 telemetry, the telemetry component will delete the configmap correct?

Reply:
if delete it, during reconcile it will be recreated. otherwise there is an issue when telemetry got disabled while approving the migration, reconcile never happened and the configmap doesn't have owner, so if stc got deleted the configmap won't be cleaned up

Reply:
add the details in the comment when possible